test/metabase/api/search_test.clj

```diff
@@ -816,3 +817,60 @@
        (testing "error if creator_id is not an integer"
          (let [resp (mt/user-http-request :crowberto :get 400 "search" :q search-term :created_by "not-a-valid-user-id")]
            (is (= {:created_by "nullable value must be an integer greater than zero."} (:errors resp)))))))))
+
+(deftest verified-filter-test
+  (let [search-term "Verified filter"]
+    (t2.with-temp/with-temp
+      [:model/Card       {v-card-id :id}    {:name (format "%s Verified Card" search-term)}
+       :model/Card       {card-id :id}      {:name (format "%s Normal Card" search-term)}
+       :model/Card       {un-v-card-id :id} {:name (format "%s Unverified Card" search-term)}
+       :model/Collection {v-coll-id :id}    {:name (format "%s Verified Collection" search-term) :authority_level "official"}
+       :model/Collection {coll-id :id}      {:name (format "%s Normal Collection" search-term)}]
+
+      (mt/with-verified-cards [v-card-id]
+        (premium-features-test/with-premium-features #{:official-collections :content-verification}
+          (testing "Able to filter only verified items"
+            (let [resp (mt/user-http-request :crowberto :get 200 "search" :q search-term :verified true)]
+
+              (testing "do not returns duplicated verified cards"
+                (is (= 1 (->> resp
+                              :data
+                              (filter #(= {:model "card" :id v-card-id} (select-keys % [:model :id])))
+                              count))))
+
+              (testing "only a subset of models are applicable"
+                (is (= #{"card" "collection"} (set (:available_models resp)))))
+
+              (testing "results contains only entities with the specified creator"
+                (is (= #{[v-coll-id  "collection" "Verified filter Verified Collection"]
+                         [v-card-id  "card"       "Verified filter Verified Card"]}
+                       (->> (:data resp)
+                            (map (juxt :id :model :name))
+                            set))))))
+
+          (testing "Able to filter non-verified items"
+            (let [resp (mt/user-http-request :crowberto :get 200 "search" :q search-term :verified false)]
+
+              (testing "only a subset of models are applicable"
+                (is (= #{"card" "collection"} (set (:available_models resp)))))
+
+              (testing "results contains only entities with the specified creator"
```

Comment:
```suggestion
              (testing "results contains only non-verified entities"
```

Reply:
Although maybe this test needs changing. See https://github.com/metabase/metabase/pull/32662#discussion_r1276370872